packages/dynamite/dynamite_runtime/lib/src/dynamite_client.dart

```diff
@@ -425,33 +428,42 @@ class DynamiteClient {
     return executeRawRequest(
       method,
       uri,
-      {...?baseHeaders, ...headers},
+      {
+        if (userAgent != null) 'user-agent': userAgent!,
+        ...?baseHeaders,
+        ...headers,
+      },
       body,
       validStatuses,
     );
   }
 
   /// Executes a HTTP request against give full [uri].
-  Future<HttpClientResponse> executeRawRequest(
+  Future<http.StreamedResponse> executeRawRequest(
     String method,
     Uri uri,
     Map<String, String> headers,
     Uint8List? body,
     Set<int>? validStatuses,
   ) async {
-    final request = await httpClient.openUrl(method, uri);
+    final request = http.Request(method, uri);
     request.headers.addAll(headers);
 
     if (body != null) {
-      request.add(body);
+      request.bodyBytes = body;
     }
+
     if (cookieJar != null) {
-      request.cookies.addAll(await cookieJar!.loadForRequest(uri));
+      final cookies = await cookieJar!.loadForRequest(uri);
+      request.headers['cookie'] = cookies.join('; ');
     }
 
-    final response = await request.close();
-    if (cookieJar != null) {
-      await cookieJar!.saveFromResponse(uri, response.cookies);
+    final response = await httpClient.send(request);
+
+    final cookieHeader = response.headers['set-cookie'];
+    if (cookieHeader != null && cookieJar != null) {
+      final cookies = cookieHeader.split(RegExp(',(?=[^ ])')).map(Cookie.fromSetCookieValue).toList();
+      unawaited(cookieJar!.saveFromResponse(uri, cookies));
```

Comment:
Should be awaited to ensure queued requests always use the correct requests.

Reply:
The api of this package is garbage (it should use FutureOr).
Cookies are not persisted on web (the browser does that for us) so it will always use the `DefaultCookieJar` which implements this in a synchronous way.
I'd like not to interrupt the execution here so I wrapped it in unawaited.

Ideally the interface would be FutureOr with both default implementations implementing this synchronous. If I feel like it I'll open an Issue/PR upstream but I doubt that they'd respond in a reasonable time.

Reply:
But for now we should still await it.

Do you think it would make sense to write our own implementation?

Reply:
The code is really complex. I'm against writing our own for now.

And awaiting it will interrupt the execution (async functions always do so. Even if they complete synchronously).

Reply:
But we need to await it to ensure the integrity of the cookies? Or am I missing something?